src/MainContainer/MainContainer.js

```diff
@@ -0,0 +1,112 @@
+import React, { Component } from "react";
+import MoviesList from "src/MoviesList";
+import Search from "src/Search";
+import SearchByType from "src/SearchByType";
+import SortByCategory from "src/SortByCategory";
+import Header from "src/Header";
+import Footer from "src/Footer";
+import cx from 'classnames';
+import globalStyles from '../../node_modules/bootstrap/dist/css/bootstrap.min.css';
+import styles from "./MainContainer.css";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+import {
+  requestApiData,
+  clickStoreData,
+} from "src/redux/actions";
+
+class MainContainer extends Component {
+  constructor(props) {
+    super(props);
+  }
+
+  filterMovies = (searchText, activeSearch, allMovies) => {
+    const currentMovieList = allMovies;
+    let filteredItem = searchText || "";
+    let filteredMovieList = [];
+    if (activeSearch === 'title') {
+      filteredMovieList = currentMovieList.filter(movie => {
+        filteredItem = filteredItem.toLowerCase();
+        return movie.title.toLowerCase().includes(filteredItem);
+      });
+    } else if(activeSearch === 'genre') {
+      filteredMovieList = currentMovieList.filter(movie => {
+        filteredItem = filteredItem.toLowerCase();
+        const matchedGenre = movie.genres.find(item => item.toLowerCase().includes(filteredItem))
+        return !!matchedGenre;
+      });
+    } 
+    return filteredMovieList;
+  };
+
+  sortRelease = (movies) => {
```

Comment:
generally all this operations could be done with actions and stored in redux store as `filteredMovies`